sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1468 @@
+However, the `foo` property will not appear on the returned `annotated_ptr` because it is a runtime property.
+
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P1{bar, baz, foo{1}};
+properties P2{bar, foo{1}};
+properties P3{bar, baz}
+
+// APtr1 is of type annotated_ptr<int*, decltype(properties{bar, baz, usm_kind<sycl::usm::alloc::device>})>
+auto APtr1 = annotated_malloc_device<int>(N, q, P1);
+
+// APtr2 is of type annotated_ptr<int*, decltype(properties{bar, usm_kind<sycl::usm::alloc::device>})>
+auto APtr2 = annotated_malloc_device<int>(N, q, P2);
+
+// APtr3 is of type annotated_ptr<int*, decltype(properties{bar, baz, usm_kind<sycl::usm::alloc::device>})>
+auto APtr3 = annotated_malloc_device<int>(N, q, P3);
+
+// Runtime properties are not present on the returned annotated_ptr
+static_assert(std::is_same_v<decltype(APtr1), decltype(APtr3)>);
+
+// APtr1 and APtr2 do not have the same properties
+static_assert(!std::is_same_v<decltype(APtr1), decltype(APtr2)>);
+----
+
+If the USM memory allocation kind is known at compile-time, the compile-time-constant property `sycl::ext::oneapi::experimental::usm_kind` will also appear on the returned `annotated_ptr`.
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+// APtr4 is of type annotated_ptr<int*, decltype(properties{usm_kind<sycl::usm::alloc::device>})>
+auto APtr4 = annotated_malloc_device<int>(N, q, properties{});
+
+// APtr5 is of type annotated_ptr<int*, decltype(properties{usm_kind<sycl::usm::alloc::host>})>
+auto APtr5 = annotated_malloc_host<int>(N, q, properties{});
+
+// APtr6 is of type annotated_ptr<int*, decltype(properties{usm_kind<sycl::usm::alloc::shared>})>
+auto APtr6 = annotated_malloc_shared<int>(N, q, properties{});
+
+static_assert(!std::is_same_v<decltype(APtr4), decltype(APtr5)>);
+static_assert(!std::is_same_v<decltype(APtr4), decltype(APtr6)>);
+static_assert(!std::is_same_v<decltype(APtr5), decltype(APtr6)>);
+----
+
+This extension also introduces USM memory allocation functions with `properties` support that allow the USM memory allocation kind to be specified at runtime.
+In this case, the returned `annotated_ptr` will not have the `sycl::ext::oneapi::experimental::usm_kind` property (unless that property is also passed in).
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P4{bar, foo{1}};
+
+// APtr7 is of type annotated_ptr<int*, decltype(properties{bar})>
+auto APtr7 = annotated_malloc<int>(N, q, sycl::usm::alloc::device, P4);
+
+----
+
+If the USM memory allocation kind specified by a parameter to the allocation function is different than the USM memory allocation kind specified by the `sycl::ext::oneapi::experimental::usm_kind` property, the USM memory allocation kind specified by the property takes precedence.
```

Comment:
I think you said you would make this an exception.

Reply:
Yes, I have now updated this to throw an exception.